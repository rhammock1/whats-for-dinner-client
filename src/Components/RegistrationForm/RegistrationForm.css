.registration-form {
  width: 75%;
  height: auto;
  margin: auto;
  padding: 5px 10px;
  background-color: #313753;
  border-radius: 5px;
  box-shadow: 2px 3px 6px 3px #121212; 
}
