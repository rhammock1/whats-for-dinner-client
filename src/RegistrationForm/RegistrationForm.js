import React from 'react';
import AuthApiService from '../services/auth-api-service';

class RegistrationForm extends React.Component {
  state = { error: null }

  handleSubmit = event => {
    event.preventDefault();
    const { user_name, first_name, password } = event.target;

    this.setState({ error: null })
    AuthApiService.postUser( {
      user_name: user_name.value,
      first_name: first_name.value,
      password: password.value,
    })
      .then(user => {
        user_name.value = ''
        first_name.value = ''
        password.value = ''
        this.props.onRegistrationSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state;
    return (
      <form className='registration-form' onSubmit={this.handleSubmit} >
        <div role='alert'>{error && <p className='red'>{error}</p>}</div>
        <div className='form-group'>
          <label htmlFor='user_name'>Username: </label>
          <input required type='text' id='user_name' name='user_name' />
        </div>
        <div className='form-group'>
          <label htmlFor='first_name'>First Name: </label>
          <input required type='text' id='first_name' name='first_name' />
        </div>
        <div className='form-group'>
          <label htmlFor='password'>Password: </label>
          <input required type='password' id='password' name='password' />
        </div>
        <button type='submit'>Register</button>
      </form>
    )
  }
}

export default RegistrationForm;