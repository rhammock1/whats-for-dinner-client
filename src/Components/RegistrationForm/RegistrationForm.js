import React from 'react';
import AuthApiService from '../../services/auth-api-service';
import TokenService from '../../services/token-service';
import './RegistrationForm.css';

class RegistrationForm extends React.Component {
  state = { error: null }

 
  handleSubmit(event) {
    event.preventDefault();
    const { user_name, first_name, password } = event.target;

    this.setState({ error: null })
    AuthApiService.postUser( {
      user_name: user_name.value,
      first_name: first_name.value,
      password: password.value,
    })
      .then(async user => {
        
        await AuthApiService.postLogin({
          user_name: user_name.value,
          password: password.value,
        })
          .then(res => {
            TokenService.saveAuthToken(res.authToken)
            TokenService.saveUserName(res.user_name)
            TokenService.saveUserId(res.id);
          })
          .catch(res => {
            this.setState({ error: res.error })
          })
        this.props.onRegistrationSuccess(user)
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state;
    return (
      <form className='registration-form' onSubmit={event => this.handleSubmit(event)} >
        <fieldset>
          <legend><h3>Sign Up</h3></legend>
          <div role='alert'>{error && <p className='red'>{error}</p>}</div>
        <div className='form-group'>
          <label htmlFor='user_name'>Username: </label>
          <input required type='text' id='user_name' name='user_name' />
        </div>
        <div className='form-group'>
          <label htmlFor='first_name'>First Name: </label>
          <input required type='text' id='first_name' name='first_name' />
        </div>
        <div className='form-group'>
          <label htmlFor='password'>Password: </label>
          <input required type='password' id='password' name='password' />
        </div>
        <button type='submit'>Register</button>
        </fieldset>
        
      </form>
    )
  }
}

export default RegistrationForm;