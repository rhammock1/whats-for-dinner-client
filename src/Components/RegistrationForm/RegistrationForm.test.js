import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RegistrationForm from './RegistrationForm';


describe('Form component for user to add new thing', () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<RegistrationForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})