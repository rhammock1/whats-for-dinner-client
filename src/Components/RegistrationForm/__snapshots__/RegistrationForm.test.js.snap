// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration form renders the complete form 1`] = `
<form
  className="registration-form"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      <h3>
        Sign Up
      </h3>
    </legend>
    <div
      role="alert"
    />
    <div
      className="form-group"
    >
      <label
        htmlFor="user_name"
      >
        Username: 
      </label>
      <input
        id="user_name"
        name="user_name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="first_name"
      >
        First Name: 
      </label>
      <input
        id="first_name"
        name="first_name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password: 
      </label>
      <input
        id="password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <button
      type="submit"
    >
      Register
    </button>
  </fieldset>
</form>
`;
